src/Veldrid/MTL/MTLCommandList.cs

```diff
@@ -956,29 +956,31 @@ private void BindBuffer(DeviceBufferRange range, uint set, uint slot, ShaderStag
                         ? slot + baseBuffer
                         : slot + _vertexBufferCount + baseBuffer);
 
-                    if (!_boundVertexBuffers.TryGetValue(index, out var boundBuffer))
+                    if (!_boundVertexBuffers.TryGetValue(index, out var boundBuffer) || boundBuffer.Buffer != range.Buffer)
```

Comment:
I mentioned in DM but these dictionary lookups and thrashing are pretty high alloc. For another day but worth mentioning since same line.

Reply:
I don't think it's these ones that you mentioned. But yes these ones can likely be removed pretty easily; the `MTLCommandBuffer` one less so.